schemas/notes.py

```diff
@@ -0,0 +1,29 @@
+from ma import ma
+from models.notes import NotesModel
+from models.user import UserModel
+from utils.time import time_format
+from marshmallow import fields, validates, ValidationError
+
+class NotesSchema(ma.SQLAlchemyAutoSchema):
+    class Meta:
+        model = NotesModel
+        include_fk = True
+        include_relationships = True
+        exclude = ["userinfo"]
+
+    created_at = fields.DateTime(time_format)
+    updated_at = fields.DateTime(time_format)
+
+    userinfo = fields.Nested("UserSchema")
+    user = fields.Method("showFullName", deserialize = "getUserId")
+    def showFullName(self, obj):
+        return "%s %s" % (obj.userinfo.firstName, obj.userinfo.lastName)
+
+    def getUserId(self, value):
+        return int(value)
```

Comment:
It seems you know more about marshmallow than I do at this point. I'm confused by the lack of decorators on the two methods above. Where are the `obj` and `value` parameters coming from?

Reply:
It's not obvious from the documentation, but ```obj``` refers to the class object passed. In most Python code, this would be ```self```. And ```value``` refers to the value in a for a given json key. So if an json object contains ```{id: 1, key: arbitrary}```, and you were over-writing the deserialization for ```id```, the ```value``` would be ```1```.